stock_available_to_promise_release/models/stock_move.py

```diff
@@ -463,31 +504,25 @@ def _after_release_assign_moves(self):
         moves._action_assign()
 
     def _release_split(self, remaining_qty):
-        """Split move and create a new picking for it.
-
-        By default, when we split a move at release to isolate the remaining qty
-        into a new move, we also create a new picking for it so that we can
-        release it later as soon as the qty is available.
-        This behavior can be changed by setting the flag no_backorder_at_release
-        on the stock.route of the move. This will allow to create the backorder
-        at the end of the picking process and release the unreleased moves into
-        the same picking as long as the picking is not done. By doing so, we
-        can also cleanup the backorders of the linked pickings created when
-        a released move was not processed (no qty found, or no time to do it for
-         example).
-        """
-        context = self.env.context
-        self = self.with_context(release_available_to_promise=True)
-        new_move = self  # Work on the current move if split doesn't occur
-        new_move_vals = self._split(remaining_qty)
-        if new_move_vals:
-            new_move = self.create(new_move_vals)
-            new_move._action_confirm(merge=False)
-        # Picking assignment is needed here because `_split` copies the move
-        # thus the `_should_be_assigned` condition is not satisfied
-        # and the move is not assigned.
-        new_move._assign_picking()
-        return new_move.with_context(**context)
+        """Split move and put remaining_qty to a backorder move."""
+        new_move_vals = self.with_context(release_available_to_promise=True)._split(
+            remaining_qty
+        )
+        new_move = self.create(new_move_vals)
+        new_move._action_confirm(merge=False)
+        return new_move
```

Comment:
Why do your remove the check `if new_move_vals`. It's safe to ensure that you've  values before calling create.